astropy/cosmology/scalar_inv_efuncs.pyx

```diff
@@ -0,0 +1,222 @@
+    double wp, double apiv, double wa):
+  cdef double opz = 1.0 + z
+  cdef Odescl = opz**(3. * (1. + wp + apiv * wa)) * exp(-3. * wa * z / opz)
+  return (opz**2 * (opz * Om0 + Ok0) + Ode0 * Odescl)**(-0.5)
+
+# Massless neutrinos
+def wpwacdm_inv_efunc_nomnu(double z, double Om0, double Ode0, double Ok0,
+    double Or0, double wp, double apiv, double wa):
+  cdef double opz = 1.0 + z
+  cdef Odescl = opz**(3. * (1. + wp + apiv * wa)) * exp(-3. * wa * z / opz)
+  return ((((opz * Or0 + Om0) * opz) + Ok0) * opz**2 +
+          Ode0 * Odescl)**(-0.5)
+
+# With massive neutrinos
+def wpwacdm_inv_efunc(double z, double Om0, double Ode0, double Ok0,
+    double Ogamma0, double NeffPerNu, int nmasslessnu, list nu_y, double wp,
+    double apiv, double wa):
+
+  cdef double opz = 1.0 + z
+  cdef double Or0 = Ogamma0 * (1.0 + nufunc(opz, NeffPerNu, nmasslessnu, nu_y))
+  cdef Odescl = opz**(3. * (1. + wp + apiv * wa)) * exp(-3. * wa * z / opz)
+  return ((((opz * Or0 + Om0) * opz) + Ok0) * opz**2 + Ode0 * Odescl)**(-0.5)
+
+######## w0wzCDM
+# No relativistic species
+def w0wzcdm_inv_efunc_norel(double z, double Om0, double Ode0, double Ok0,
+    double w0, double wz):
+  cdef double opz = 1.0 + z
+  cdef Odescl = opz**(3. * (1. + w0 - wz)) * exp(-3. * wz * z)
+  return (opz**2 * (opz * Om0 + Ok0) + Ode0 * Odescl)**(-0.5)
+
+# Massless neutrinos
+def w0wzcdm_inv_efunc_nomnu(double z, double Om0, double Ode0, double Ok0,
+    double Or0, double w0, double wz):
+  cdef double opz = 1.0 + z
+  cdef Odescl = opz**(3. * (1. + w0 - wz)) * exp(-3. * wz * z)
+  return ((((opz * Or0 + Om0) * opz) + Ok0) * opz**2 +
+          Ode0 * Odescl)**(-0.5)
+
+# With massive neutrinos
+def w0wzcdm_inv_efunc(double z, double Om0, double Ode0, double Ok0,
+    double Ogamma0, double NeffPerNu, int nmasslessnu, list nu_y, double w0,
+    double wz):
+
+  cdef double opz = 1.0 + z
+  cdef double Or0 = Ogamma0 * (1.0 + nufunc(opz, NeffPerNu, nmasslessnu, nu_y))
+  cdef Odescl = opz**(3. * (1. + w0 - wz)) * exp(-3. * wz * z)
+  return ((((opz * Or0 + Om0) * opz) + Ok0) * opz**2 + Ode0 * Odescl)**(-0.5)
+
+######## Neutrino relative density function
+# Scalar equivalent to FLRW.nu_realative_density in core.py
+#  Please see that for further discussion.
+# This should only be called with massive neutrinos (e.g., nu_y is not empty)
+# Briefly, this is just a numerical fitting function to the true relationship,
+#  which is too expensive to want to evaluate directly.  The
+#  constants which appear are:
+#    p = 1.83  -> numerical fitting constant from Komatsu et al.
+#  1/p = 0.54644... -> same constant
+#    k = 0.3173 -> another fitting constant
+#  7/8 (4/11)^(4/3) = 0.2271... -> fermion/boson constant for neutrino
+#                                   contribution -- see any cosmology book
+#  The Komatsu reference is: Komatsu et al. 2011, ApJS 192, 18
+
+cdef nufunc(double opz, double NeffPerNu, int nmasslessnu, list nu_y):
```

Comment:
Yeah it doesn't make sense *at all* in that case that it would be slower.  Even with bounds checking (which is going to be happening with a list too, among other things).  Did you make sure to include a
```
cdef unsigned int i
```
to make sure the loop variable is a C integer?  I often forget that the first time around.